### ISSUE TITLE
Incorrect warnings about NewType

### ISSUE THREAD

**Issue by e2t** [python/mypy#7678]
```from typing import NewType
Mass = NewType('Mass', float)
x: Mass = Mass(97.28)
x += Mass(2.29)
```
**mypy** reports about last line:

> error: Incompatible types in assignment (expression has type "float", variable has type "Mass")

It is very strange.

**Comment by JukkaL** [comment]
Unfortunately this is a just how NewTypes work. They don't change the signatures of `__add__` etc. to return the NewType. All methods (other than `__init__`) default to returning whatever the original definition returns (in this case, `float` from `float.__add__`).

To work around the issue you can use something like `x = Mass(x + 2.29)`. If you want more type safety, you have to create a real class.
